src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -329,18 +329,34 @@ internal BoundExpression BindValueAllowArgList(ExpressionSyntax node, Diagnostic
                 return null;
             }
 
+            SymbolKind containingMemberOrLambdaKind = this.ContainingMemberOrLambda.Kind;
+            Debug.Assert(containingMemberOrLambdaKind == SymbolKind.Field || containingMemberOrLambdaKind == SymbolKind.Method);
+            bool isMemberInitializer = containingMemberOrLambdaKind == SymbolKind.Field;
```

Comment:
Initializers on properties are always bound in context of backing field. Right from the start.